### PR TITLE
Support archive downloads on oc10 backends aka archiver v1.0.0

### PR DESCRIPTION
**Pull request by kulmann**
## Description
Support archive downloads on oc10 backends aka archiver v1.0.0.

Also moved archive download related code to the `archiverService` to keep it
in one place and expose less api for the service.
Unit tests have been changed to only use the public api of the service
as well.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/6239

## Motivation and Context
customer demand

## How Has This Been Tested?

- unit tests + manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/src/index.js [hunk 1]
```diff
@@ -127,7 +127,14 @@ export default {
   userReady({ store }) {
     archiverService.initialize(
       store.getters.configuration.server || window.location.origin,
-      get(store, 'getters.capabilities.files.archivers', [])
+      get(store, 'getters.capabilities.files.archivers', [
```
Comment (kulmann) [inline packages/web-app-files/src/index.js:130]: long term we should add the capability to `core` / php based backend. Added it as a fallback for compatibility in the meantime.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/24280/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesActionMenu-fileFolderActionMenu_feature-L37.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/index.js (hunk 1 above)
Comment (pascalwengerter) [reply packages/web-app-files/src/index.js:130]: True. Care to open a ticket in github.com/owncloud/core/issues?:)

**Review by pascalwengerter — APPROVED** [review]
Code LGTM and CI is happy, however as seen in the screenshare it doesn't work on public links due to the lack of token/presigned URL

**Comment by kulmann** [comment]
> Code LGTM and CI is happy, however as seen in the screenshare it doesn't work on public links due to the lack of token/presigned URL

will followup in a separate PR in the future, but needs backend changes first. for the moment this already brings new functionality for the authenticated context.

**Pull request merged**